### PR TITLE
Fix #1986: Migrate Travis CI to GitHub Actions

### LINKED ISSUE #1986: Migrate Travis CI to GitHub Actions
## Issue Description

Type: *CI*

### Describe what happened (or what feature you want)

Migrate Travis CI to [GitHub Actions](https://github.com/alibaba/Sentinel/actions), as Travis CI org will be shutting down soon.

Contributions are welcomed!

### PR DESCRIPTION
**Pull request by brothelul**
### Describe what this PR does / why we need it
Migrate Travis CI to GitHub Actions 

### Does this pull request fix one issue?

Fix #1986 

### Describe how you did it
migrate the old .travis.yml to .github/workflows/ci.yml

### Describe how to verify it
check the Actions

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Sentinel CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (sczyh30) [inline .github/workflows/ci.yml:7]: Maybe we could choose an LTS version (e.g. 18.04)? FYI: https://github.com/actions/virtual-environments/issues/1816

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        uses: actions/checkout@v2
+
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+          architecture: x64
+
+      - name: Build with Maven
+        run: mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V -DminimumPriority=1
```
Comment (sczyh30) [inline .github/workflows/ci.yml:22]: Here we'll need to execute both build and test.

FYI (config in Travis CI): https://docs.travis-ci.com/user/languages/java/#maven-dependency-management

**Review by sczyh30 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (brothelul) [reply .github/workflows/ci.yml:22]: OK, I will change it later.

**Review by sczyh30 — APPROVED** [review]
LGTM

**Comment by sczyh30** [comment]
Thanks for contributing!

**Pull request merged**